### PR TITLE
When getting all pages, if the API response is an object, wrap it in an array before pushing it to results.

### LINKED ISSUE #267: Support endpoints that return objects in _requestAllPages
Currently, if you call `_requestAllPages` on an endpoint that returns an object, like `/search/`, you won't get any results back.

This is due to the handling of the result object - it's assumed it will be an array.

See: https://github.com/michael/github/blob/master/github.js#L124

### PR DESCRIPTION
**Pull request by jeffstieler**
Fixes #267.

### CODE REVIEW

File: github.js [hunk 1]
```diff
@@ -121,6 +121,10 @@
                   return cb(err);
                }
 
+               if (Object.prototype.toString.call(res) === '[object Object]') {
```
Comment (AurelioDeRosa) [inline github.js]: Is there any reason why you haven't used:

```js
if (typeof res === 'object')
```

Comment (jeffstieler) [inline github.js]: `typeof` returns `'object'` for arrays, see: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof

Comment (AurelioDeRosa) [inline github.js]: Right, I didn't think about it. A better suggestion is:

```js
if (!(res instanceof Array))
```

**Comment by AurelioDeRosa** [comment]
Hi @jeffstieler. Would you mind update your PR?

**Comment by jeffstieler** [comment]
@AurelioDeRosa sorry this slipped through the cracks - updated.

**Comment by AurelioDeRosa** [comment]
Merged.  Thank you.

**Pull request closed without merging**